SpatialGDK/Source/SpatialGDKFunctionalTests/Private/SpatialFunctionalTest.cpp

```diff
@@ -190,6 +191,63 @@ int ASpatialFunctionalTest::GetNumberOfClientWorkers()
 	return Counter;
 }
 
+void ASpatialFunctionalTest::AddActorDelegation_Implementation(AActor* Actor, uint8 ServerWorkerId, bool bPersistOnTestFinished /*= false*/)
+{
+	ISpatialFunctionalTestLBDelegationInterface* DelegationInterface = GetDelegationInterface();
+
+	if (DelegationInterface != nullptr)
+	{
+		bool bAddedDelegation = DelegationInterface->AddActorDelegation(Actor, ServerWorkerId, bPersistOnTestFinished);
+		ensureMsgf(bAddedDelegation, TEXT("Tried to delegate Actor %s to Server Worker %d but couldn't"), *Actor->GetName(), ServerWorkerId);
+	}
+}
+
+void ASpatialFunctionalTest::RemoveActorDelegation_Implementation(AActor* Actor)
+{
+	ISpatialFunctionalTestLBDelegationInterface* DelegationInterface = GetDelegationInterface();
+
+	if (DelegationInterface != nullptr)
+	{
+		bool bRemovedDelegation = DelegationInterface->RemoveActorDelegation(Actor);
+		ensureMsgf(bRemovedDelegation, TEXT("Tried to remove Delegation from Actor %s but couldn't"), *Actor->GetName());
+	}
+}
+
+bool ASpatialFunctionalTest::HasActorDelegation(AActor* Actor, uint8& WorkerId, bool& bIsPersistent)
+{
+	WorkerId = 0;
+	bIsPersistent = 0;
+
+	ISpatialFunctionalTestLBDelegationInterface* DelegationInterface = GetDelegationInterface();
+
+	bool bHasDelegation = false;
+
+	if (DelegationInterface != nullptr)
+	{
+		VirtualWorkerId AuxWorkerId;
+
+		bHasDelegation = DelegationInterface->HasActorDelegation(Actor, AuxWorkerId, bIsPersistent);
+
+		WorkerId = static_cast<uint8>(AuxWorkerId);
+	}
+
+	return bHasDelegation;
+}
+
+ISpatialFunctionalTestLBDelegationInterface* ASpatialFunctionalTest::GetDelegationInterface() const
+{
+	USpatialNetDriver* SpatialNetDriver = Cast<USpatialNetDriver>(GetNetDriver());
+	if (SpatialNetDriver)
+	{
+		ULayeredLBStrategy* LayeredLBStrategy = Cast<ULayeredLBStrategy>(SpatialNetDriver->LoadBalanceStrategy);
+		if(LayeredLBStrategy != nullptr)
+		{
+			return Cast<ISpatialFunctionalTestLBDelegationInterface>(LayeredLBStrategy->GetLBStrategyForVisualRendering());
```

Comment:
Hmm, I see now why you wanted the ability to switch out the LayeredLBStrategy. Maybe a comment somewhere that this Actor delegation will not work properly with offloading?

Reply:
It would be great to have real world complex LB scenarios to figure out what to do here.

Reply:
Also, I really think that we should be able to override which class is spawned for ULayeredLBStrategy. Then this mechanism would be handled at the root.